halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/DeploymentController.java

```diff
@@ -22,69 +22,76 @@
 import com.netflix.spinnaker.halyard.model.v1.Halconfig;
 import lombok.Data;
 import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.*;
 
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
 import java.util.List;
 import java.util.stream.Collectors;
 
 @RestController
+@NoArgsConstructor
 @RequestMapping("/v1/config/deployments")
 public class DeploymentController {
   @Autowired
   HalconfigParser halyardConfig;
 
-  @RequestMapping(value = "/{deployment:.+}", method = RequestMethod.GET)
-  DeploymentConfiguration deploymentConfigurations(@PathVariable String deployment) throws UnrecognizedPropertyException {
+  public DeploymentConfiguration getDeploymentConfiguration(String deployment) throws UnrecognizedPropertyException {
     Halconfig halconfig = halyardConfig.getConfig();
-    if (halconfig != null) {
-      List<DeploymentConfiguration> matching = halconfig.getDeploymentConfigurations()
-          .stream()
-          .filter(d -> d.getName().equals(deployment))
-          .collect(Collectors.toList());
+    List<DeploymentConfiguration> matching = halconfig.getDeploymentConfigurations()
+        .stream()
+        .filter(d -> d.getName().equals(deployment))
+        .collect(Collectors.toList());
```

Comment:
https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/Iterables.html#getOnlyElement-java.lang.Iterable- has a neat example of using `MoreCollectors.onlyElement()`